[Release] All integrations (#17260)
DataDog/integrations-core: sqlserver/CHANGELOG.md

<!-- towncrier release notes start -->

## 17.1.0 / 2024-03-22

***Added***:

> * update custom_queries configuration to support optional collection_interval ([#16957](https://github.com/DataDog/integrations-core/pull/16957))

Reviewer: ```suggestion
* Update custom_queries configuration to support optional collection_interval ([#16957](https://github.com/DataDog/integrations-core/pull/16957))
```

> * Tag sqlserver agent queries with service:datadog-agent ([#17162](https://github.com/DataDog/integrations-core/pull/17162))

Reviewer 2: ```suggestion
* Tag SQL Server Agent queries with service:datadog-agent ([#17162](https://github.com/DataDog/integrations-core/pull/17162))
```

***Fixed***:

* Skip index usage collection for tempdb, fixes a blocking query ([#16977](https://github.com/DataDog/integrations-core/pull/16977))
* Fix an issue where logging of obfuscation errors would still hide the statement when it's part of a stored procedure. If the stored procedure obfuscation fails, the `procedure_signature` tag will be set to `__procedure_obfuscation_error__`. ([#17020](https://github.com/DataDog/integrations-core/pull/17020))
* Skip `sqlserver.latches.latch_wait_time metric` on SQL Server version 2012 and 2014. ([#17063](https://github.com/DataDog/integrations-core/pull/17063))
* Update the configuration to include the `metric_prefix` option ([#17065](https://github.com/DataDog/integrations-core/pull/17065))

> * Improve service check of autodiscovered databases for the sqlserver integration. The fix prevents agent from creating a connection per database.

Reviewer 2: ```suggestion
* Improve service check of autodiscovered databases for the SQL Server integration. The fix prevents agent from creating a connection per database.
```
